src/EVM/Format.hs

```diff
@@ -828,7 +828,7 @@ hexByteString :: String -> ByteString -> ByteString
 hexByteString msg bs =
   case BS16.decodeBase16Untyped bs of
     Right x -> x
-    _ -> internalError $ "invalid hex bytestring for " ++ msg
+    _ -> error $ "Invalid hex bytestring provided for '" ++ msg ++ "'. Bytestring provided: '" ++ show bs ++ "'"
```

Comment:
this change is a clear upgrade to existing code so totally fine for now, but just a general comment not a great ux to use `error` for user facing errors, and this function might be better written to return an `Either` (or perhaps we could add some special `Error` effect to `Effects.hs`?)

Reply:
OK, I have re-written the error handling here to be MUCH better. It's done with `Maybe` and we handle the errors at the source when the `Nothing` happens, and give very user-friendly error messages, with `exitFailure`, etc. @d-xo  let me know what you think!